### PR TITLE
Remove typing_extensions and Delete advanced-overload-generic

### PR DESCRIPTION
**Pull request by F-park**
As `ParamSpec` add to `typing` in 3.10, `typing_extensions` is unnecessary.

### CODE REVIEW

**Comment by F-park** [comment]
the test failed not related this commit, because upgrade for pyright can detect more error

File: challenges/advanced-overload-generic/solution.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def foo(value: Foo[str]) -> str:
 
 
 @overload
-def foo(value: Foo[T]) -> Foo[T]:
+def foo(value: Foo[list]) -> Foo[list]:
```
Comment (laike9m) [inline challenges/advanced-overload-generic/solution.py:26]: Why this change? since it requires "otherwise returns a Foo[T]"

Comment (F-park) [reply challenges/advanced-overload-generic/solution.py:26]: it make the test failed, just fix the test.

Comment (laike9m) [reply challenges/advanced-overload-generic/solution.py:26]: Could this be because of you upgraded pyright's version in this PR? I wouldn't recommend bundling unrelated changes together.

Also, if a test fails, changing the test instead of finding the root cause is not good.

Comment (F-park) [reply challenges/advanced-overload-generic/solution.py]: upgraded `pyright` cause the test falled, and I revert "fix test".

**Comment by laike9m** [comment]
Let me know when it's ready

**Comment by F-park** [comment]
it is ready for review.

**Comment by laike9m** [comment]
The test is still failing. Seems pyright's version has not been reverted

image

**Comment by F-park** [comment]
Not relate, ci install `pyright` from pdm.

https://github.com/laike9m/Python-Type-Challenges/blob/main/.github/workflows/test.yaml#L18-L19

https://github.com/laike9m/Python-Type-Challenges/pull/107/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L11

I consider modify to `pyright<1.1.348` is a bad choice.

**Comment by laike9m** [comment]
Can confirm this issue happened because of a [behavior change](https://github.com/microsoft/pyright/releases/tag/1.1.339) in 1.1.339

> Added a new typeCheckingMode called "standard". It's a subset of "strict" but a superset of "basic". It is the new default mode, and it should cover all of the required checks for conformance with the Python typing standard.

And after some trials & errors & searching, my conclusion is that this is impossible to achieve the desired behavior without using some wordarounds (adding `# type: ignore[misc]`), or writing the actual implementation to return correct types. The core issue is that Python's type system does not support "excluding certain types"

Thus, I think we can delete this challenge to unblock this PR. You can delete it in this PR or a separate one, both works

**Comment by F-park** [comment]
I delete `advanced-overload-generic`, and the unit test pass.
